designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/CustomActionValidator.scala

```diff
@@ -0,0 +1,160 @@
+package pl.touk.nussknacker.ui.validation
+
+import cats.data.Validated.Invalid
+import io.circe.{Decoder, Encoder, HCursor, Json}
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.PartSubGraphCompilationError
+import pl.touk.nussknacker.engine.api.deployment.CustomActionCommand
+import pl.touk.nussknacker.engine.api.parameter.ParameterName
+import pl.touk.nussknacker.engine.deployment.{
+  CustomActionDefinition,
+  CustomActionParameter,
+  CustomActionValidationResult
+}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.restmodel.CustomActionRequest
+import pl.touk.nussknacker.ui.BadRequestError
+
+class CustomActionValidator(val allowedActions: List[CustomActionDefinition]) {
```

Comment:
Lets rename to ScenarioActionParameterValidator, we are aiming here to validation at "action parameter" level. And because of that I recommend to pass single action properties like this:
```
class ScenarioActionParameterValidator(actionParameters: Map[String, CustomActionParameter])
```

Reply:
Changed to injecting single aciton into the constructor. I think the naming is ok for now, we can clean up when the change to action parameter comes